### ISSUE TITLE
Ideas for improvement

### ISSUE THREAD

**Issue by tacsipacsi** [MatmaRex/dtstats#1]
Some ideas for improvement:
* Include a link to this repo. Since the README contains important information, this is important so that people have a chance to read that information. Unlike the below bullet points, I believe this is a must-have.
* Think small – on any screen narrower than 1600px, the table of contents and the graphs overlap; on screens narrower than 1500px, the graphs overflow horizontally. On any screen shorter than 900px, the end of the table of contents is inaccessible. There is always room for improvement, but making the TOC hideable/collapsible and making it have its own collapsible box seems to be a good effort/impact ratio.
* Tabbed/collapsible interface? This could be an alternative to fixing the table of contents. Maybe tabs if there’s JavaScript and TOC otherwise, somewhat similarly to MediaWiki preferences. Or checkboxes to select which one or more graphs should appear – this would allow comparing two wikis that are alphabetically far from each other (e.g. I regularly edit huwiki, Commons, Wikidata etc., how do they compare to each other?).
* Start all graphs at the same time. Even if some wikis got DiscussionTools later, this would help comparison by lining up the data.

Please note that I didn’t look at the source code (I’m not fluent in Ruby), so do tell me if any of my ideas is technically too difficult.

**Comment by tacsipacsi** [comment]
Also, you could link to the tool in the repository details (website field), making the link more visible.

**Comment by MatmaRex** [comment]
>Include a link to this repo. Since the README contains important information, this is important so that people have a chance to read that information. Unlike the below bullet points, I believe this is a must-have.

Done

>Start all graphs at the same time. Even if some wikis got DiscussionTools later, this would help comparison by lining up the data.

Done

>Also, you could link to the tool in the repository details (website field), making the link more visible.

Done

**Comment by MatmaRex** [comment]
For the other visual improvements, I think the first step might be to rewrite the visualization using a different program/library. I made it with `gnuplot`, just for fun (I haven't used it before), and it turns out that it's not ideal for charts on websites, presumably because it was invented at a time when you would probably be making charts for print ;)

To that end, the data for the charts is now available at https://dtstats.toolforge.org/data.tgz, and I am hoping someone will do it for me ;)

**Comment by tacsipacsi** [comment]
Thanks! I couldn’t get `gnuplot` work (as I wrote in the commit message of 5441629ca3d00e1ba2fd2e69d285650855765c44), but hopefully someone else will. By the way, if you want others to do it for you, maybe you should add some developer information – I guessed what to do based on the `make.bat` file, but the `gnuplot` issue may be just because I didn’t do exactly what you did, instead of some Linux vs Windows difference. Maybe the full paths to the checked-out https://github.com/wikimedia/operations-mediawiki-config should also be configurable…

**Comment by MatmaRex** [comment]
> Thanks! I couldn’t get `gnuplot` work (as I wrote in the commit message of [5441629](https://github.com/MatmaRex/dtstats/commit/5441629ca3d00e1ba2fd2e69d285650855765c44)), but hopefully someone else will.

I think that's a shell quoting problem, `gnuplot` actually expects literal `'` characters there, and presumably your shell interprets them itself while mine doesn't. I could fix that, but I actually found a better way that doesn't need funny quoting: ad7af6141816aec34c466c8f2437070f4cc6fb54

**Comment by MatmaRex** [comment]
> By the way, if you want others to do it for you, maybe you should add some developer information – I guessed what to do based on the `make.bat` file, but the `gnuplot` issue may be just because I didn’t do exactly what you did, instead of some Linux vs Windows difference. Maybe the full paths to the checked-out https://github.com/wikimedia/operations-mediawiki-config should also be configurable…

Well, that's just the first problem, next you'll find that the querying depends on having access to [WMF Analytics replicas](https://wikitech.wikimedia.org/wiki/Data_Engineering/Systems/MariaDB), and probably also on my SSH configuration (although it's pretty standard)…

The SQL query itself should work just fine on [Toolforge replicas](https://wikitech.wikimedia.org/wiki/Help:Toolforge/Database) too, but the mechanics of running queries differ, and I've never used those.

You might also find that your `gnuplot` produces slightly different output than mine. I had some issues with fonts when making this, and I hacked up some margins to work around that.

It's just a lot more effort to make the scripts work everywhere, rather than just on my machine, and this was supposed to be a fun weekend project… ;)

**Comment by tacsipacsi** [comment]
> I think that's a shell quoting problem, `gnuplot` actually expects literal `'` characters there, and presumably your shell interprets them itself while mine doesn't. I could fix that, but I actually found a better way that doesn't need funny quoting: [ad7af61](https://github.com/MatmaRex/dtstats/commit/ad7af6141816aec34c466c8f2437070f4cc6fb54)

Yes, that error is gone. It still doesn’t run, but that’s just because you used the `set key offset` feature introduced in [5.4.5](http://www.gnuplot.info/ReleaseNotes_5_4_5.html), while [my gnuplot, packaged by Debian](https://packages.debian.org/bullseye/gnuplot), is two years old (like just about anything Debian packages…). But I could just install it from Git if I wanted.
